components/Extensions/tests/Element/FrameworkElementExtensionsTests.RelativeAncestor.cs

```diff
@@ -0,0 +1,154 @@
+    public void FrameworkElementExtension_RelativeAncestor_InDataTemplate(FrameworkElementRelativeAncestorDataTemplateTestPage page)
+    {
+        var list = page.FindDescendant<ListView>();
+
+        Assert.IsNotNull(list, "Couldn't find listview");
+
+        int count = 0;
+        foreach (var item in list.FindDescendants().OfType<TextBlock>())
+        {
+            count++;
+            Assert.AreEqual("Hello", item.Text, "Text didn't match binding of ancestor tag property");
+        }
+
+        Assert.AreEqual(3, count, "Didn't find three textblocks");
+    }
+
+    [TestCategory("FrameworkElementExtension")]
+    [UIThreadTestMethod]
+    public async Task FrameworkElementExtension_RelativeAncestor_FreeParentBaseline(FrameworkElementRelativeAncestorDataTemplateTestPage page)
+    {
+        var text = page.FindDescendant<TextBox>();
+
+        Assert.IsNotNull(text, "Couldn't find TextBox");
+
+        // Grab a hold of a weak reference for TextBox so we can detect when it unloads.
+        WeakReference textRef = new(text);
+        text = null;
+
+        var parent = page.FindDescendant<Grid>();
+
+        Assert.IsNotNull(parent, "Couldn't find parent Grid");
+
+        // Remove all the children from the grid to simulate it unloading.
+        VisualTreeHelper.DisconnectChildrenRecursive(parent);
+        parent.Children.Clear();
+        parent = null;
+
+        // Wait for the Visual Tree to perform removals and clean-up
+        await CompositionTargetHelper.ExecuteAfterCompositionRenderingAsync(() => { });
+
+        // Wait for the .NET Garbage Collector to clean up references
+        GC.Collect();
+        GC.WaitForPendingFinalizers();
+
+        Assert.IsFalse(textRef.IsAlive, "TextBox is still alive...");
+    }
+
+    [TestCategory("FrameworkElementExtension")]
+    [UIThreadTestMethod]
+    public async Task FrameworkElementExtension_RelativeAncestor_FreeParent(FrameworkElementRelativeAncestorDataTemplateTestPage page)
+    {
+        var list = page.FindDescendant<ListView>();
+
+        Assert.IsNotNull(list, "Couldn't find listview");
+
+        // Grab a hold of a weak reference for ListView so we can detect when it unloads.
+        WeakReference listRef = new(list);
+        list = null;
+
+        // Remove all the children from the grid to simulate it unloading.
+        VisualTreeHelper.DisconnectChildrenRecursive(page);
+        page.Content = null;
+
+        // Wait for the Visual Tree to perform removals and clean-up
+        await CompositionTargetHelper.ExecuteAfterCompositionRenderingAsync(() => { });
+
+        // Wait for the .NET Garbage Collector to clean up references
+        GC.Collect();
+        GC.WaitForPendingFinalizers();
+
+        Assert.IsFalse(listRef.IsAlive, "ListView is still alive...");
+    }
+
+    [TestCategory("FrameworkElementExtension")]
+    [UIThreadTestMethod]
+    public async Task FrameworkElementExtension_RelativeAncestor_FreePageNavigation()
```

Comment:
_continued from https://github.com/CommunityToolkit/Windows/issues/107#issuecomment-1619338414_

@sylveon thanks for the comment. Yes, the test at the top is just looking at the random `TextBox` to test the testing methodology.

The other two tests here try unloading the ListView directly from the tree and doing a page navigation. Even without the `Unloaded` code I added here though, I wasn't seeing the `TextBlock` holding up the unload and seeing the `ListView` still freed.

Is there something else I could be missing here in this test setup?

Reply:
Maybe there's something different with the C# GC, but with a 1:1 C++ translation I was definitely seeing memory leaks.

Reply:
Since the test is working as intended but still unable to reproduce the issue, we'll keep #107 open for now and move future work to a future PR.